src/RFC-0304_DanGlossary.md

```diff
@@ -0,0 +1,91 @@
+|-------------|-------------------------------------|--------------------------------------------------------------------------------------------|
+| Component   | $H(TxHash \mathbin\Vert OutputNum)$ | An instance of a type defined by a template. Arbitrary state with the notion of ownership. |
+| Resource    | $H(TxHash \mathbin\Vert OutputNum)$ | A value container for a single token. Operations: Mint, Deposit, Withdraw, Burn, Update.   |
+| KeyValueMap | $H(TxHash \mathbin\Vert OutputNum)$ | A lazy key value map owned by a component. Entries must scale.                             |
+
+### Template level
+
+| Term        | Key                       | Description                                                                                                                                                                                                                               | 
+|-------------|---------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Vault       | TBD, <ResourceAddress, n> | Thinly wraps a single resource and provides convenient API for resource transfer, balances etc.                                                                                                                                           |
+| Bucket      | locally unique number     | A temporary resource container. All buckets should be consumed by the end of the transaction.                                                                                                                                             |
+| Worktop     | --                        | A temporary object container for working data while the transaction is processed. All resource containers (buckets, vaults, etc.) should be cleared from the workspace at the end of the transaction or the transaction is not committed. |
+| AccessRules | TBD                       | Access rules that limit operations on components and resources.                                                                                                                                                                           |
+| Badge       | TBD, <ResourceAddress, n> | A resource used with AccessRules and to generate bearer tokens                                                                                                                                                                            |
+| BearerToken | nonce                     | A delegated, partially stateless proof providing limited and scoped access to a particular resource or component                                                                                                                          |
+
```

Comment:
Add a section for Base Layer terms:

VN Merkle Tree - (Alternatively EpochTree, EpochCommitment, EpochVnsTree)